pkg/parsers/operatingsystem/operatingsystem_linux.go

```diff
@@ -18,15 +23,27 @@ var (
 
 // GetOperatingSystem gets the name of the current operating system.
 func GetOperatingSystem() (string, error) {
-	b, err := ioutil.ReadFile(etcOsRelease)
+	osReleaseFile, err := os.Open(etcOsRelease)
 	if err != nil {
 		return "", err
 	}
-	if i := bytes.Index(b, []byte("PRETTY_NAME")); i >= 0 {
-		b = b[i+13:]
-		return string(b[:bytes.IndexByte(b, '"')]), nil
+	defer osReleaseFile.Close()
+
+	scanner := bufio.NewScanner(osReleaseFile)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "PRETTY_NAME=") {
+			data := strings.SplitN(line, "=", 2)
+			prettyNames, err := shellwords.Parse(data[1])
+			if err != nil || len(prettyNames) != 1 {
+				return "", fmt.Errorf("PRETTY_NAME is invalid")
```

Comment:
If we have a value for 'err' I'd prefer if we returned that to the user so they know what we don't like their string.
Likewise, if len != 1 then we should say that they can't have more than one value, or they need to use quotes if they have spaces.
Minor things, but I'd prefer to be as helpful as possible since the current error message doesn't give a clue as to why we think its invalid.